components/style/rule_tree/mod.rs

```diff
@@ -533,9 +533,14 @@ impl StrongRuleNode {
     }
 
     fn next_sibling(&self) -> Option<WeakRuleNode> {
-        // FIXME(emilio): Investigate what ordering can we achieve without
-        // messing things up.
-        let ptr = self.get().next_sibling.load(Ordering::SeqCst);
+        // Note that relaxed is fine because the tree is only iterated while
+        // mutated for search/insertion. In that case, the only thing that may
+        // happen is that we find a null sibling pointer when other thread has
+        // written already to that memory location.
+        //
```

Comment:
I think this basically hinges on the outcome of the discussion below. If that write needs to be Release, then this read needs to be Acquire, otherwise the synchronization doesn't take effect.